### PR TITLE
Bump node-problem-detector to v0.8.5

### LINKED ISSUE #96746: node-problem-detector does not discover OOMs in newer Linux kernels
**What happened**:
In a cluster with NPD addon installed and Linux 5.1+ as the kernel OS of the nodes OOMs are not being captured by the NPD.

**What you expected to happen**:
NPD should inform about OOMs by emitting appropriate K8s events.

**How to reproduce it (as minimally and precisely as possible)**:
Create a K8s cluster with NPD addon and Linux 5.1+ installed on the nodes. Schedule a pod which leaks memory. Try to list events with `reason: OOMKilling` field after an OOM occurs to see there's none.

**Anything else we need to know?**:
Fix is already merged into newest (v0.8.5) NPD release: https://github.com/kubernetes/node-problem-detector/pull/481.

We're currently waiting for the release of Docker image of NPD v0.8.5 - we can proceed with fixing this in k/k afterwards. After it gets merged to the master branch, it should be then cherry-picked to older releases as well.

**Environment**:
Any Kubernetes version with NPD addon installed and Linux as the node OS with the kernel version 5.1 or newer.

### PR DESCRIPTION
**Pull request by tosi3k**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
NPD v0.8.0 doesn't detect OOMs in newer Linux kernels. v0.8.5 contains a fix that enables the NPD to detect them in both older and newer kernels.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes/kubernetes/issues/96746.

**Special notes for your reviewer**:
Holding since the Docker image for NPD v0.8.5 is not released yet.
/hold
/sig scalability
/assign @wojtek-t

SHA512 digest of the v0.8.5 tar package:
```
zawodny@zawodny  ~/Downloads  sha512sum node-problem-detector-v0.8.5.tar.gz
3fbf97a38c06d8fcc8c46f956a6e90aa1b47cb42d50ddcfd1a644a7e624e42ee523db2f81e08fbfb21b80142d4bafdbedce16e8b62d2274a5b2b703a56d9c015  node-problem-detector-v0.8.5.tar.gz
```

**Does this PR introduce a user-facing change?**:

```release-note
Bump node-problem-detector version to v0.8.5 to fix OOM detection in with Linux kernels 5.1+
```

### CODE REVIEW

File: cluster/addons/node-problem-detector/npd.yaml
Comment (invidian) [inline cluster/addons/node-problem-detector/npd.yaml:50]: It seems this image (or `k8s.gcr.io/node-problem-detector/node-problem-detector:v0.8.5`) is not yet available.

Comment (tosi3k) [reply cluster/addons/node-problem-detector/npd.yaml:50]: Yup, that's why I put this PR on hold in the first place :).

**Comment by wojtek-t** [comment]
@tosi3k - please open k/k issue for this one - this is a bug and I want this to land in 1.20 (and we should then cherrypick back)

**Comment by tosi3k** [comment]
> @tosi3k - please open k/k issue for this one - this is a bug and I want this to land in 1.20 (and we should then cherrypick back)

Done - I updated the PR's description with the URL to the issue.

**Comment by wojtek-t** [comment]
@tosi3k - was the image already built?

**Comment by tosi3k** [comment]
> @tosi3k - was the image already built?

Not yet, unfortunately :/. I will unhold the PR (and rerun the tests) once it's released.

**Comment by tosi3k** [comment]
v0.8.5 is released:
```
 ~  docker pull k8s.gcr.io/node-problem-detector/node-problem-detector:v0.8.5
v0.8.5: Pulling from node-problem-detector/node-problem-detector
39fafc05754f: Already exists 
4ace085f408a: Already exists 
c9131edd1592: Pull complete 
7245b8152cc0: Pull complete 
f688a05461d1: Pull complete 
d5cffb7d7176: Pull complete 
Digest: sha256:07645e26231454292ddcd20863e8fe1a9422d0e9888bd1a1ff3189695e143138
Status: Downloaded newer image for k8s.gcr.io/node-problem-detector/node-problem-detector:v0.8.5
k8s.gcr.io/node-problem-detector/node-problem-detector:v0.8.5
```

/hold cancel

**Comment by tosi3k** [comment]
@wojtek-t PTAL.

**Comment by wojtek-t** [comment]
/lgtm
/approve

Adding to milestone as this is fixing a bug.
/milestone v1.20

**Comment by wojtek-t** [comment]
@tosi3k - once this merges, please cherrypick to previous releases

**Review by kemooo3333 — APPROVED** [review]

**Comment by ialidzhikov** [comment]
> @tosi3k - once this merges, please cherrypick to previous releases

v1.17 -#96783
v1.18 -#96784
v1.19 -#96785

**Pull request merged**
